File: week2/陈志荣/GC_percentage.py
#!/usr/bin/python
with open('E:/github/bioinformatics-programming-homework/week2/陈志荣/chr1.fa', 'r') as f:
	str=f.read()

str=str.lower()
str_list=list(str)
dict={}

for i in str:
	if i in dict:
		count=dict[i]
	else:
		count=0
	count=count+1
	dict[i]=count

GC=dict['c']+dict['g']
total=dict['c']+dict['g']+dict['a']+dict['t']
GC_per=GC/total
print("GC percentage is:",GC_per)

#GC percentage is: 0.417242924908526
#[Finished in 162.8s]
#
#
#
#
#
#
#
#{'>': 1, 'c': 48055044, 'h': 1, 'r': 1, '1': 1, '\n': 4979130, 'n': 18475410, 't': 67244164, 'a': 67070277, 'g': 48111528}
#
#sep=''
#str=sep.join(str)
#resoult={}
#for i in str:
#    resoult[i]=str.count(i)
#print(resoult)
#

#str_list = ['Python','Tab']
#a = ''
#print (a.join(str_list))


#print("Loadding data")
#try:
#    f = open('E:/github/bioinformatics-programming-homework/week2/陈志荣/chr1.fa', 'r')    # 打开文件
#    data = f.read()                   # 读取文件内容
#finally:
#    if f:
#        f.close()                     # 确保文件被关闭

#or :
#with open('E:/github/bioinformatics-programming-homework/week2/陈志荣/chr1.fa', 'r') as f:
#    data= f.read()
#data=data.lower()
#print(type(data))
#resoult={}
#for i in data:
#    resoult[i]=data.count(i)
#print("result",resoult)

#

#def count_GC(input):
#	for i in input:
#		res={}
#		

File: week2/陈志荣/GC_percentage.R
# 编程练习第二周（2019.01.21-2019.01.26）
# 任务：
# 计算人类1号染色体的GC含量
# 
# 输入：人类hg38版本基因组中1号染色体的序列（fasta格式）
# 输出：GC含量
# GC含量是在所研究的对象（例如放线菌）的全基因组中，（鸟嘌呤）（Guanine）和胞嘧啶（Cytosine）所占的比例。
# 
# 输入文件：
# cd week2
# chr1.fa.gz

setwd("E:\\github\\bioinformatics-programming-homework\\week2\\陈志荣")
library(data.table)#data.table::fread can read large data quickly
library(stringr)#str_count
help(fread)
chr1<- fread("chr1.fa")
chr1<-paste(chr1$`>chr1`,collapse = "")
chr1<-str_replace_all(chr1, "N", "")
G_C=sum(attr(gregexpr("G|C",chr1,ignore.case=TRUE)[[1]], "match.length"))
G_C_per=paste(round(G_C/str_length(chr1),2)*100,"%")
print(paste("GC percentage:",G_C_per))
